flask_app/FlaskApp.py

```diff
@@ -1,16 +1,152 @@
+
+    def create_and_insert_to_database_video_metadata_frame_models(self,
+                                                                  path: str,
+                                                                  image_list: list,
+                                                                  video_file_name: str):
+        """
+        Create and insert to database video metadata and frame models.
+        :param path: Video path.
+        :param image_list: Image list.
+        :param video_file_name: Video file name.
+        """
+        observation = self._get_observation_name_from_path(path)
+        video_from_db = self.db.get_video(observation, video_file_name, len(image_list))
+
+        if not video_from_db:
+            self.db.insert_one(data_models.Video(observation_name=observation,
+                                                 os_path=video_file_name,
+                                                 number_of_frames=len(image_list)))
+
+            video_model = self.db.get_video(observation_name=observation,
+                                            os_path=video_file_name,
+                                            number_of_frames=len(image_list))
+
+            frame_model_list = frame_utils.create_frame_models(len(image_list), video_model.id, video_file_name)
+
+            self.create_and_insert_metadata_and_set_metadata_id_to_frames(frame_model_list, image_list)
+
+            self.db.insert_many(frame_model_list)
+
+    def _get_observation_name_from_path(self, path: str):
+        """
+        Extract observation name from path.
+        :param path: Video file path.
+        :return: Observation name.
+        """
+        return os.path.basename(path).split('_')[0]
+
+    def create_and_insert_metadata_and_set_metadata_id_to_frames(self, frames: list, images: list):
+        """
+        Create and insert metadata and set metadata id to frames.
+        :param frames: Frame list.
+        :param images: Image list.
+        """
+        for index, image in enumerate(images):
+            fov, azi, elev, tag = self.get_metadata_arguments(image)
+            metadata = self.db.get_metadata(fov, azi, elev, tag)
+
+            if not metadata:
+                self.db.insert_one(data_models.Metadata(fov=fov, azimuth=azi, elevation=elev, tag=tag))
+                metadata = self.db.get_metadata(fov, azi, elev, tag)
+
+            frames[index].set_metadata_id(metadata.id)
+
+    def get_metadata_arguments(self, image: list):
+        """
+        Extract fov, azi, elev, tag parameters from image.
+        :param image: Image.
+        :return: fov, azi, elev, tag
+        """
+        fov, azi, elev = metadata_utils.generate_metadata(image)
+        tag = metadata_utils.is_frame_tagged(image)
+
+        return fov, azi, elev, tag
+
+    async def upload_binary_file_to_azure(self, path: str, video_file_name: str, azure_path: str):
```

Comment:
You are uploading a video or a binary file ?

Reply:
Also why not having this method inside the azure handler?
If the method has azure in its name it should be here.